dlib/dnn/layers_abstract.h

```diff
@@ -4087,6 +4166,152 @@ namespace dlib
     template <long diag, long num, long den, typename SUBNET>
     using tril_diag = add_layer<tril_<diag, void, num, den>, SUBNET>;
 
+// ----------------------------------------------------------------------------------------
+
+    class positional_encodings_
```

Comment:
This class is already in this file on master (so don't add another copy of it) :D

Reply:
I removed this duplicated declaration